networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsRpcFuluDeprecationTest.java

```diff
@@ -0,0 +1,320 @@
+  private final StubMetricsSystem metricsSystem = new StubMetricsSystem();
+  private String protocolId;
+  private UInt64 fuluForkFirstSlot;
+  private DataStructureUtil dataStructureUtil;
+  private Spec spec;
+  private int maxBlobsPerBlock;
+
+  @BeforeEach
+  public void setup() {
+    spec =
+        TestSpecFactory.createMinimalWithCapellaDenebElectraAndFuluForkEpoch(
+            UInt64.ZERO, UInt64.ZERO, UInt64.ONE, fuluForkEpoch);
+    dataStructureUtil = new DataStructureUtil(spec);
+    messageSchema =
+        spec.atEpoch(fuluForkEpoch)
+            .getSchemaDefinitions()
+            .toVersionFulu()
+            .orElseThrow()
+            .getBlobSidecarsByRootRequestMessageSchema();
+    fuluForkFirstSlot = spec.computeStartSlotAtEpoch(fuluForkEpoch);
+    final RpcEncoding rpcEncoding =
+        RpcEncoding.createSszSnappyEncoding(spec.getNetworkingConfig().getMaxPayloadSize());
+    protocolId = BeaconChainMethodIds.getBlobSidecarsByRootMethodId(1, rpcEncoding);
+    maxBlobsPerBlock =
+        SpecConfigDeneb.required(spec.forMilestone(SpecMilestone.ELECTRA).getConfig())
+            .getMaxBlobsPerBlock();
+
+    when(peer.approveRequest()).thenReturn(true);
+    when(peer.approveBlobSidecarsRequest(eq(callback), anyLong()))
+        .thenReturn(allowedObjectsRequest);
+    reset(combinedChainDataClient);
+    when(combinedChainDataClient.getBlockByBlockRoot(any()))
+        .thenReturn(
+            SafeFuture.completedFuture(
+                Optional.of(dataStructureUtil.randomSignedBeaconBlock(fuluForkFirstSlot))));
+    // epoch 1 = electra is finalized, epochs 2+ not
+    when(combinedChainDataClient.getFinalizedBlock())
+        .thenReturn(
+            Optional.of(
+                dataStructureUtil.randomSignedBeaconBlock(
+                    spec.computeStartSlotAtEpoch(UInt64.ZERO))));
+    when(combinedChainDataClient.getStore()).thenReturn(store);
+    when(combinedChainDataClient.getRecentChainData()).thenReturn(recentChainData);
+    when(callback.respond(any())).thenReturn(SafeFuture.COMPLETE);
+
+    // mock store
+    when(store.getGenesisTime()).thenReturn(genesisTime);
+    // current epoch is deneb fork epoch + 1
+    when(store.getTimeSeconds())
+        .thenReturn(
+            spec.getSlotStartTime(
+                fuluForkEpoch.increment().times(spec.getSlotsPerEpoch(ZERO)), genesisTime));
+  }
+
+  @Test
+  public void byRangeShouldSendToPeerPreFuluBlobSidecarsOnly() {
+    final BlobSidecarsByRangeMessageHandler handler =
+        new BlobSidecarsByRangeMessageHandler(spec, metricsSystem, combinedChainDataClient);
+    final UInt64 slotsPerEpoch = UInt64.valueOf(spec.getSlotsPerEpoch(ZERO));
+    // request for over 2 epochs from epoch 1 2nd slot to end of epoch 2. On epoch 2, the first slot
+    // is where fulu is activated. Expect replies limited up to (not including) fulu activation slot
+    UInt64 startSlot = UInt64.ONE.times(slotsPerEpoch).increment();
```

Comment:
I'd use something like `spec.computeStartSlotAtEpoch(ONE)` for cleanliness. Also, please, add finals whenever possible